tests/networks/nets/test_vit.py

```diff
@@ -18,45 +18,49 @@
-                                        for classification in [False, True]:
-                                            for nd in (2, 3):
-                                                test_case = [
-                                                    {
-                                                        "in_channels": in_channels,
-                                                        "img_size": (img_size,) * nd,
-                                                        "patch_size": (patch_size,) * nd,
-                                                        "hidden_size": hidden_size,
-                                                        "mlp_dim": mlp_dim,
-                                                        "num_layers": num_layers,
-                                                        "num_heads": num_heads,
-                                                        "proj_type": proj_type,
-                                                        "classification": classification,
-                                                        "num_classes": num_classes,
-                                                        "dropout_rate": dropout_rate,
-                                                    },
-                                                    (2, in_channels, *([img_size] * nd)),
-                                                    (2, (img_size // patch_size) ** nd, hidden_size),
-                                                ]
-                                                if nd == 2:
-                                                    test_case[0]["spatial_dims"] = 2  # type: ignore
-                                                    if classification:
-                                                        test_case[0]["post_activation"] = False  # type: ignore
-                                                if test_case[0]["classification"]:  # type: ignore
-                                                    test_case[2] = (2, test_case[0]["num_classes"])  # type: ignore
-                                                TEST_CASE_Vit.append(test_case)
+TEST_CASE_Vit = [
+    (
+        [
+            {
+                "in_channels": params["in_channels"],
+                "img_size": (params["img_size"],) * params["nd"],
+                "patch_size": (params["patch_size"],) * params["nd"],
+                "hidden_size": params["hidden_size"],
+                "mlp_dim": params["mlp_dim"],
+                "num_layers": params["num_layers"],
+                "num_heads": params["num_heads"],
+                "proj_type": params["proj_type"],
+                "classification": params["classification"],
+                "num_classes": params["num_classes"],
+                "dropout_rate": params["dropout_rate"],
+                **({"spatial_dims": 2} if params["nd"] == 2 else {}),
+                **({"post_activation": False} if params["nd"] == 2 and params["classification"] else {}),
```

Comment:
I feel this usage of ** with a bracketed expression isn't so readable, so this whole statements may be better implemented with a for-loop so you can construct the dictionary with actual `if` statements. You can also use the `{**params,...}` pattern here to reduce the line count a lot.